src/index.js

```diff
@@ -1,35 +1,36 @@
 import calculateFPS from './calculate-fps'
 
-export default function collectFPS (...args) {
-  const customMaxFrames = typeof args[0] === 'number'
-
-  const maxFrames = customMaxFrames ? args[0] : 10
-  const cb = customMaxFrames ? args[1] : args[0]
-  const frames = []
-
-  if (!window.requestAnimationFrame) {
-    cb(new Error('requestAnimationFrame is not available'))
-    return
+export default function collectFPS (theWindow = window) {
+  if (!theWindow.requestAnimationFrame) {
+    throw new Error('requestAnimationFrame is not available')
   }
 
+  // Collect visibility change events
   const visibilityEvents = []
 
   const onVisibilityChange = () =>
     visibilityEvents.push([!document.hidden, Date.now()])
 
   document.addEventListener('visibilitychange', onVisibilityChange)
 
-  const update = () => {
-    frames.push(Date.now())
+  // Collect frames from requestAnimationFrame
```

Comment:
I can't comment on that line, because Github sucks, but you should probably have tests for the visibility change as well. Since you're passing in the window, it should be simple enough to add.